index.js

```diff
@@ -5,308 +5,76 @@
-            // safe value for HTML attributes and HTTP headers.
-            createToken(36, 'base64', function (err, token) {
-                if (err) {
-                    return callback(err);
-                }
-                try {
-                    Object.defineProperty(res.locals, 'cspToken', {
-                        value: token,
-                        enumerable: true
-                    });
-                    callback(null, res);
-                } catch(e) {
-                    callback(new Error('Unable to set the nonce token to res.locals.'));
-                }
-            });
-        } else {
-            callback(null, res);
-        }
-    }
-
-    // middleware
-    app.use(function (req, res, next) {
-        setNonce(res, cspPolicies.policy, cspPolicies.reportPolicy, function(err, res) {
-            if (err) {
-                next(err);    
-            }
-            setHeaders(res);
-            next();
-        });
-    });
-
-
     app.signScript = function (script) {
-        appSignedScripts[sign(script)] = true;
+        csp.signedScripts[csp.sign(script)] = true;
     };
-
+    
     app.response.signScript = function (script) {
-        var scripts = getSignedScripts(this);
-        scripts[sign(script)] = true;
-        setHeaders(this);
+        var scripts = csp.getSignedScripts(this);
+        scripts[csp.sign(script)] = true;
+        csp.setContentSecurityHeaders(this);
     };
 
     app.signStyle = function (style) {
-        appSignedStyles[sign(style)] = true;
+        csp.signedStyles[csp.sign(style)] = true;
     };
 
     app.response.signStyle = function (style) {
-        var styles = getSignedStyles(this);
-        styles[sign(style)] = true;
-        setHeaders(this);
+        var styles = csp.getSignedStyles(this);
+        styles[csp.sign(style)] = true;
+        csp.setContentSecurityHeaders(this);
     };
-
+    
     app.response.setPolicy = function (config) {
         Object.freeze(config);
-        var policy = config.policy,
-            reportPolicy = config.reportPolicy;
+
+        var policy       = config.policy;
+        var reportPolicy = config.reportPolicy;
+
         try {
-            Object.defineProperty(this.locals, "cspPolicies", {
-                value: getCSPPolicies(config),
+            Object.defineProperty(this.locals, 'cspPolicies', {
+                value: csp.getCSPPolicies(config),
                 enumerable: true
             });
-        } catch(e) {
-            throw new Error('The res.locals.cspPolicies value must only be set once per request and must always be set through the `setPolicy` method.');
+        } catch (ex) {
+            throw new Error('The `res.locals.cspPolicies` value must only be set once per request and must always be set through the `setPolicy` method.');
         }
+
         if (this.locals.cspToken) {
-            setHeaders(this);
+            csp.setContentSecurityHeaders(this);
```

Comment:
This method could potentially be invoked 4 times per response.  Shouldn't this only ever be invoked once on the middleware below: `app.use`